snapshots.js

```diff
@@ -2,45 +2,134 @@ const fs = require('fs');
+ * @returns {Promise<String[]>} Array of example file paths, relative to `EXAMPLES_RELATIVE_DIR`.
+ */
+async function getExampleFiles(dir = EXAMPLES_RELATIVE_DIR) {
+  let files = [];
+  let absDir = path.join(process.cwd(), dir);
+  const entries = await fs.promises.readdir(absDir, {withFileTypes: true});
+
+  for (let entry of entries) {
+    const fullPath = path.join(absDir, entry.name);
+    if (entry.isDirectory()) {
+      if (fs.existsSync(`${fullPath}/combined.html`)) {
+        // Combined HTML file found. Assume it combines all examples in the directory and subdirectories.
+        files.push(path.join(dir, entry.name, 'combined.html'));
+      } else {
+        // Get examples from the subdirectory
+        files.push(...(await getExampleFiles(path.join(dir, entry.name))));
+      }
+    } else if (entry.isFile() && entry.name.endsWith('.html') && !entry.name.startsWith('_')) {
+      // Ignore partials & non-HTML files
+      files.push(path.join(dir, entry.name));
+    }
+  }
+
+  // Return the relative paths
+  return files.map((file) => file.replace(`${EXAMPLES_RELATIVE_DIR}/`, ''));
 }
 
-function getExampleUrls(files) {
+/**
+ * Converts example relative paths to example URLs
+ * @param fileNames {String[]} - Array of file names (relative to `EXAMPLES_RELATIVE_DIR`) to generate URLs for.
+ * @returns {String[]} Array of URLs to snapshot.
+ */
+function getExampleUrlsFromExamplePaths(fileNames) {
   // add standalone versions of the examples in base and patterns folders
-  const standaloneFiles = files.filter((file) => file.startsWith('base/') || file.startsWith('patterns/')).map((file) => path.join('standalone', file));
+  const standaloneFiles = fileNames.filter((file) => file.startsWith('base/') || file.startsWith('patterns/')).map((file) => path.join('standalone', file));
 
-  files = files.concat(standaloneFiles);
+  fileNames = fileNames.concat(standaloneFiles);
 
   // map the file paths to URLs
-  return files.map((file) => {
+  return fileNames.map((file) => {
     file = file.replace('.html', '');
-    return DOCS_URL + file;
+    return EXAMPLES_BASE_URL + file;
   });
 }
 
+/**
+ * Given an example path, returns the color themes to snapshot.
+ * @param filepath {String} Path (relative to `EXAMPLES_RELATIVE_DIR`) of the example file.
+ * @returns {String[]} Array of color theme names to snapshot.
+ *  These must match color theme query parameter names accepted by example-tools.js.
+ */
+function getThemesForExample(filepath) {
+  return ['light', 'dark', 'paper'];
+}
+
+/**
+ * Given an example path, return the widths to snapshot.
+ * @param filepath {String} Path (relative to `EXAMPLES_RELATIVE_DIR`) of the example file.
+ * @returns {Promise<Number[]>} Array of widths to snapshot, from smallest to largest.
+ */
+async function getWidthsForExample(filepath) {
+  let widths = new Set([
+    375, // Mobile
+    1280, // Desktop
+  ]);
+
+  if (filepath.endsWith('combined/')) {
```

Comment:
I don't exactly understand what is the purpose of this code. What changes if the combined path is provided?
Isn't "combined" just an HTML file? In here it seems to be treated as directory.

Reply:
Changed param names & added documentation to make this more clear, let me know if that is sufficient @bartaz